### PR TITLE
MGDAPI-69_Parameterize_namespace

### PR DESCRIPTION
**Pull request by CathalOConnorRH**
# Description
Update namespace prefix and parameterize it across all code references

To Validate:
1. Set NAMESPACE_PREFIX in make file to either namespace prefix
2. Install operator and confirm all namespaces have the new prefix. 

## Type of change
- [ ] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: pkg/controller/subscription/subscription_controller_test.go [hunk 1]
```diff
@@ -295,44 +295,44 @@ func TestShouldReconcileSubscription(t *testing.T) {
 		},
 		{
 			Name:      "Not in reconcile name list",
-			Namespace: "redhat-rhmi-operator",
+			Namespace: integreatlyv1alpha1.RHMI{}.Spec.NamespacePrefix + "operator",
```
Comment (laurafitzgerald) [inline pkg/controller/subscription/subscription_controller_test.go:298]: could we move this as a var to the top as with the other tests?

File: pkg/products/monitoring/dashboards/criticalSLOAlerts.go [hunk 2]
```diff
@@ -444,8 +448,8 @@ const MonitoringGrafanaDBCriticalSLOAlertsJSON = `{
 			"scopedVars": {
 				"product": {
 					"selected": false,
-					"text": "redhat-rhmi-3scale|ThreeScale",
-					"value": "redhat-rhmi-3scale|ThreeScale"
+					"text": "` + integreatlyv1alpha1.RHMISpec{}.NamespacePrefix + `3scale|ThreeScale",
```
Comment (laurafitzgerald) [inline pkg/products/monitoring/dashboards/criticalSLOAlerts.go:451]: same here and with others. I won't comment on the others tho. 👍

File: pkg/products/monitoring/dashboards/criticalSLOAlerts.go [hunk 3]
```diff
@@ -505,8 +509,8 @@ const MonitoringGrafanaDBCriticalSLOAlertsJSON = `{
...
+					"value": "` + integreatlyv1alpha1.RHMISpec{}.NamespacePrefix + `3scale|ThreeScale"
```
Comment (KevFan) [inline pkg/products/monitoring/dashboards/criticalSLOAlerts.go:513]: `integreatlyv1alpha1.RHMISpec{}.NamespacePrefix` doesn't look to resolve in grafana json either :thinking: 
Screenshot from 2020-09-30 13-00-15

**Review by KevFan — COMMENTED** [review]
Looks good :+1: 

Though when verifying, `integreatlyv1alpha1.RHMI{}.Spec.NamespacePrefix` didn't look to resolve for the `MiddlewareMonitoringPodCount` alert and the critical slo dashboards :thinking:

File: pkg/products/monitoring/prometheusRules.go [hunk 4]
```diff
@@ -198,7 +199,7 @@ func (r *Reconciler) newAlertsReconciler() resources.AlertReconciler {
 							"sop_url": resources.SopUrlAlertsAndTroubleshooting,
 							"message": "Pod count for namespace {{ $labels.namespace }} is {{ $value }}. Expected exactly 7 pods.",
 						},
-						Expr:   intstr.FromString("(1 - absent(kube_pod_status_ready{condition='true',namespace='redhat-rhmi-middleware-monitoring-operator'})) or sum(kube_pod_status_ready{condition='true',namespace='redhat-rhmi-middleware-monitoring-operator'}) != 7"),
+						Expr:   intstr.FromString("(1 - absent(kube_pod_status_ready{condition='true',namespace='" + integreatlyv1alpha1.RHMISpec{}.NamespacePrefix + "middleware-monitoring-operator'})) or sum(kube_pod_status_ready{condition='true',namespace='" + integreatlyv1alpha1.RHMISpec{}.NamespacePrefix + "middleware-monitoring-operator'}) != 7"),
```
[the remark this answers is not in the document]
Comment (CathalOConnorRH) [reply pkg/products/monitoring/prometheusRules.go]: @KevFan I've rewritten this so it should now resolve correctly, Can you retest when you get a chance?

File: pkg/products/monitoring/dashboards/criticalSLOAlerts.go (hunk 3 above)
Comment (CathalOConnorRH) [reply pkg/products/monitoring/dashboards/criticalSLOAlerts.go]: same as above

**Pull request merged**